spec/config_spec.rb

```diff
@@ -246,11 +247,5 @@
       local.input = :foo
       expect(local.input).to eq(:foo)
     end
-
-    it 'modifies input itself' do
```

Comment:
If you delete this spec, I'd expect it to be replaced with some other one that checks that Readline is properly initialized, by whichever code does that now,

Reply:
what about these specs?
https://github.com/dgutov/pry/blob/e1eeed547b1cebcc323bd1b78b94aec4f0dc3c12/spec/completion_spec.rb#L89-L95
we added in same PR that made the change to `send()`.

Reply:
i think this test was testing for something else, indirectly, something to do with the default for lazy_readline not being called due to `Pry.config.input = RbReadline` in .pryrc.

Reply:
Nah, these specs only test the completer. Readline calls the completer, but not vice versa.

It's basically impossible to write a high-level test for this behavior in Readline (or at least I don't know how), so I check the assigned value.

Reply:
hum okay. should be possible to test that another way, just:
```ruby
pry = Pry.new(input: Readline)
`expect(pry.input.basic_word_....).to eq("..")`.
```
?

Reply:
Sure. Except, this would be better as the second line:

```rb
expect(Readline.basic_word_....).to eq("..")
```

Reply:
What do you think: https://github.com/pry/pry/pull/1638/commits/9a6608bf43c1429565012865a2f3da2da8924ad3

It will break on jruby but specs are already broken there, so can fix at same time.

Reply:
Hm, any particular reason to call the external Ruby?

Reply:
Mostly a historic thing. Readline is a heavy external dependency that is console/terminal focused compared to `input: StringIO.new("..")` which is enough for some tests. So Readline-specific are kept there, to avoid bringing Readline into spec's that don't need it. I really doubt the spec suite lives by this though.

Reply:
OK, if you think that's best.

Reply:
Passing everywhere but jruby since https://github.com/pry/pry/pull/1638/commits/9d1bda8c86c01de2922f1c0f267c0985ecb4880f

Reply:
`require "pry"` does not implicitly require "Readline" since v0.10. It's only required if `Default#lazy_readline` is hit. So yeah, I think it's sorta better this way.